python/metashape_workflow_functions.py

```diff
@@ -1276,3 +1295,60 @@ def get_written_paths(self, as_json: bool = False):
             return json_str
         # Otherwise just return the dictionary representation
         return self.written_paths
+
+    def get_cameraset_origin(self, round: int = 100) -> Metashape.Vector:
+        """
+        Goes through the EXIF lat/lon data from the cameras, converts it into
+        the project CRS, and then reports the (rounded) camera mean as the
+        project origin. The purpose is to get around the accuracy limitations
+        of float32 values (the default for point clouds and meshes) by having
+        a fixed origin offset per project.
+
+        NOTE: The shifted origin is reported by exportModel via the
+        save_metadata_xml mechanism.
+
+        Arguments:
+            round (int): The value to round (really floor) the origin to. The
+                purpose of this is to make the origin more human readable,
+                while still getting the accuracy benefits that come when your
+                float32 data points are values in the 100s-1000s instead of in
+                the millions
+
+        Returns: Metashape.Vector of the camera origin. For now Z is kept at 0
+            and only (X, Y) are calculated from the cameras.
+        """
+
+        # The camera reference location is known to be in lat/lon
+        camera_crs = Metashape.CoordinateSystem("EPSG::4326")
+
+        # Average the camera locations without using libraries like numpy
+        x = 0.0
+        y = 0.0
+        invalid = 0
```

Comment:
nit: might be slightly more intuitive to just define `n_valid` here and increment it appropriately.